segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/security/TLSHelper.java

```diff
@@ -37,34 +38,40 @@
      *
      * @param pathToCertificateFile the path to the PEM-encoded server certificate file
      * @param pathToServerKeyFile the path to the PEM-encoded file containing the server's encrypted private key
+     * @param tlsProtocolVersion the version of the TLS protocol
      * @return a {@link SslContext} built from the specified {@code pathToCertificateFile} and {@code pathToServerKeyFile}
      * @throws NullPointerException if either {@code pathToCertificateFile} or {@code pathToServerKeyFile} is null
      * @throws IllegalArgumentException if either {@code pathToCertificateFile} or {@code pathToServerKeyFile} is empty
      * @throws RuntimeException if there is a failure in building the {@link SslContext}
      */
-    public static SslContext newServerSslContext(String pathToCertificateFile, String pathToServerKeyFile) {
+    public static SslContext newServerSslContext(String pathToCertificateFile, String pathToServerKeyFile, String tlsProtocolVersion) {
         Exceptions.checkNotNullOrEmpty(pathToCertificateFile, "pathToCertificateFile");
         Exceptions.checkNotNullOrEmpty(pathToServerKeyFile, "pathToServerKeyFile");
-        return newServerSslContext(new File(pathToCertificateFile), new File(pathToServerKeyFile));
+        Exceptions.checkArgument(TLSProtocolVersion.parse(tlsProtocolVersion) != null, tlsProtocolVersion, "Invalid TLS Protocol Version");
+        return newServerSslContext(new File(pathToCertificateFile), new File(pathToServerKeyFile), tlsProtocolVersion);
     }
 
     /**
      * Creates a new instance of {@link SslContext}.
      *
      * @param certificateFile the PEM-encoded server certificate file
      * @param serverKeyFile the PEM-encoded file containing the server's encrypted private key
+     * @param tlsProtocolVersion version of TLS protocol
      * @return a {@link SslContext} built from the specified {@code pathToCertificateFile} and {@code pathToServerKeyFile}
      * @throws NullPointerException if either {@code certificateFile} or {@code serverKeyFile} is null
      * @throws IllegalStateException if either {@code certificateFile} or {@code serverKeyFile} doesn't exist or is unreadable.
      * @throws RuntimeException if there is a failure in building the {@link SslContext}
      */
-    public static SslContext newServerSslContext(File certificateFile, File serverKeyFile) {
+    public static SslContext newServerSslContext(File certificateFile, File serverKeyFile, String tlsProtocolVersion) {
         Preconditions.checkNotNull(certificateFile);
         Preconditions.checkNotNull(serverKeyFile);
+        Preconditions.checkNotNull(tlsProtocolVersion);
```

Comment:
Why don't you do the validation here? You parse this string above just to verify it's valid, then below you parse it yet again so you can get the protocols. 

I suggest you get rid of the extra validation above and just do it here - only once. You can have it throw the `IllegalArgEx` here if it can't be parsed.

Reply:
Fixed.Removed the extra validation.